frontend/packages/console-shared/src/components/quick-search/QuickSearchModalBody.tsx

```diff
@@ -33,20 +37,68 @@ const QuickSearchModalBody: React.FC<QuickSearchModalBodyProps> = ({
   searchPlaceholder,
   allCatalogItemsLoaded,
   icon,
+  maxDimension,
+  viewContainer,
 }) => {
   const [catalogItems, setCatalogItems] = React.useState<CatalogItem[]>(null);
   const [catalogTypes, setCatalogTypes] = React.useState<CatalogType[]>([]);
+  const [maxHeight, setMaxHeight] = React.useState(60);
+  const [minHeight, setMinHeight] = React.useState(60);
+  const [minWidth, setMinWidth] = React.useState(60);
   const [searchTerm, setSearchTerm] = React.useState<string>(
     getQueryArgument('catalogSearch') || '',
   );
   const [selectedItemId, setSelectedItemId] = React.useState<string>('');
   const [selectedItem, setSelectedItem] = React.useState<CatalogItem>(null);
   const [viewAll, setViewAll] = React.useState<CatalogLinkData[]>(null);
-  const listCatalogItems =
-    limitItemCount > 0 ? catalogItems?.slice(0, limitItemCount) : catalogItems;
+  const [items, setItems] = React.useState<number>(limitItemCount);
+  const [modalSize, setModalSize] = React.useState<{ height: number; width: number }>();
+  const [draggableBoundary, setDraggableBoundary] = React.useState<string>(null);
   const ref = React.useRef<HTMLDivElement>(null);
   const fireTelemetryEvent = useTelemetry();
 
+  const getModalHeight = () => {
+    let height: number = 60;
+    const itemsHeight = viewAll?.length ? 388 : 365;
+    if (catalogItems?.length > 0) {
+      if (modalSize?.height >= minHeight) {
+        return modalSize?.height;
+      }
+      height += itemsHeight + (viewAll?.length - 1) * 23;
+    }
+    return height;
+  };
+
+  const listCatalogItems = React.useMemo(
+    () => (limitItemCount && limitItemCount > 0 ? catalogItems?.slice(0, items) : catalogItems),
+    [items, catalogItems, limitItemCount],
+  );
+
+  React.useEffect(() => {
+    if (viewContainer) {
+      const className = viewContainer.className.replace(' ', '.');
+      setDraggableBoundary(`.${className}`);
+    }
+  }, [viewContainer]);
```

Comment:
Hmmmm... it's unfortunate that the rnd lib doesn't take an element 🤔 

Suggested changes since this is probably the only way we can do this:

- `classList` is a list of class names, probably would lead to a smoother implementation (you're using `replace`, and not `replaceAll`, so you could lead to an unexpected result)
- This feels like a memo if you wanted to stop the computation on each run -- which although I'm unsure if that's hugely needed since it'll likely be just a join of strings

The only alternative I can think of is we accept a selector instead of an HTMLElement at the top and if we need the HTMLElement we can do the `querySelector` internally.

Reply:
how about if I do:
``` if (viewContainer) {
      const className = viewContainer.classList;
      setDraggableBoundary(`.${className[0]}`);
    }